### ISSUE TITLE
We need to find a way to manually refresh the next.js cache

### ISSUE THREAD

**Issue by dadiorchen** [Greenstand/treetracker-web-map-client#1277]
I just encountered this problem:
1. visit https://beta-map.treetracker.org/organizations/fcctrees it return 404, at this moment it is correct, because we don't have this org
2. Then create organization by name: fcctrees
3. visit the URL again, it is still 404
4. Only way to refresh is to restart the next.js

We need to find a way to trigger the refresh and correct the 404 failure.

---

Some hints:

- Please read our [readme](https://github.com/Greenstand/treetracker-web-map-client#treetracker-web) for more information/guide/tutorial.
- For design guidelines checkout our [Figma design](https://www.figma.com/file/XdYFdjlsHvxehlrkPVYq0l/Greenstand-Webmap).
- Here is [an engineering book](https://greenstand.gitbook.io/engineering/) in Greenstand.
- To know more about our organization, visit our [website](https://greenstand.org).
- If you want to join the slack community (some resources need the community member's permission), please leave your email address.
- More issues in this project: https://github.com/orgs/Greenstand/projects/26

**Comment by okdv** [comment]
@dadiorchen ah yes, if i remember correctly we could create an endpoint that would manually refresh the cache for a specified page. there may also be a way to not cache the page at all if it routes to a 404 or something- but im not sure, i would need to look into that some more

been away working on a private contract so havent had much time for side stuff lately, but that contract should be done in about a week and i'll be back contributing some more :smile: let me look into this in the meantime and see what i can find

**Comment by keyy123** [comment]
@okdv will you be back soon? 😟

**Comment by keyy123** [comment]
@dadiorchen I don't mind giving this ticket a try. I'll ask in slack for additional details to solve issues as they appear.

**Comment by keyy123** [comment]
The issue at first glance seems to be one where we need to establish a cache with revalidation on demand.

**Comment by okdv** [comment]
#1020 is the PR I had for this a while ago but forgot to fix the wrong HTTP error code

just updated that and submitted the PR for review

sorry holiday season has been crazier than expected :smile:

**Comment by dadiorchen** [comment]
Actually the revalidate time now is 24 hours, we need to be so so we get better performance, this is just the difficulty we need to solve, setting a cache in normal, but need a special way to kick out it and refresh the page.

**Comment by okdv** [comment]
@dadiorchen makes sense for the cache timing. Does that endpoint satisfy the needs as far as cache revalidation goes? Or should there be some additional functionality to it?

**Comment by dadiorchen** [comment]
This is still a valid issue, here is @okdv 's work https://github.com/Greenstand/treetracker-web-map-client/pull/1020
we need someone who can continue and finish this for us.
